### PR TITLE
[HDInsight] Support Node Reboot feature

### PR DESCRIPTION
**Pull request by aim-for-better**
Add two cmdlets:
1. Get-AzHDInsightHost to Get the hosts of the cluster
2. Restart-AzHDInsightHost to restart the specific hosts of the cluster

The related cmdlets' design review is [here](https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/627) 

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/HDInsight/HDInsight/Az.HDInsight.psd1 [hunk 1]
```diff
@@ -133,7 +134,7 @@ PrivateData = @{
         # IconUri = ''
 
         # ReleaseNotes of this module
-        ReleaseNotes = '* Supported Customer-managed key disk encryption.'
+        ReleaseNotes = '* Supported list hosts and restart hosts of the cluster'
```
Comment (VeryEarly) [inline src/HDInsight/HDInsight/Az.HDInsight.psd1:137]: release note will be handled every release, please revert this line

File: src/HDInsight/HDInsight/ManagementCommands/GetAzureHDInsightHostCommand.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        private const string SetByResourceIdParameterSet = "SetByResourceIdParameterSet";
+        private const string SetByInputObjectParameterSet = "SetByInputObjectParameterSet";
+
+        [Parameter(
+            Position = 0,
+            Mandatory = false,
+            ParameterSetName = SetByNameParameterSet,
+            HelpMessage = "Gets or sets the name of the resource group.")]
+        [ResourceGroupCompleter]
+        public string ResourceGroupName { get; set; }
```
Comment (VeryEarly) [inline src/HDInsight/HDInsight/ManagementCommands/GetAzureHDInsightHostCommand.cs:40]: ResourceGroup should be mandatory
https://github.com/Azure/azure-rest-api-specs/blob/92e846e0135bce5f13b37330562714ecb9eaa058/specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/virtualMachines.json#L54

File: src/HDInsight/HDInsight/help/Get-AzHDInsightHost.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
+---
+external help file: Microsoft.Azure.PowerShell.Cmdlets.HDInsight.dll-Help.xml
+Module Name: Az.HDInsight
+online version:
```
Comment (VeryEarly) [inline src/HDInsight/HDInsight/help/Get-AzHDInsightHost.md:4]: please add online version

File: src/HDInsight/HDInsight/help/Restart-AzHDInsightHost.md [hunk 4]
```diff
@@ -0,0 +1,203 @@
+---
+external help file: Microsoft.Azure.PowerShell.Cmdlets.HDInsight.dll-Help.xml
+Module Name: Az.HDInsight
+online version:
```
Comment (VeryEarly) [inline src/HDInsight/HDInsight/help/Restart-AzHDInsightHost.md:4]: please add online version

Comment (aim-for-better) [reply src/HDInsight/HDInsight/help/Restart-AzHDInsightHost.md]: Added

---
In reply to: [438670785](https://github.com/Azure/azure-powershell/pull/12124#discussion_r438670785) [](ancestors = 438670785)

File: src/HDInsight/HDInsight/help/Get-AzHDInsightHost.md (hunk 3 above)
Comment (aim-for-better) [reply src/HDInsight/HDInsight/help/Get-AzHDInsightHost.md]: Added

---
In reply to: [438670540](https://github.com/Azure/azure-powershell/pull/12124#discussion_r438670540) [](ancestors = 438670540)

File: src/HDInsight/HDInsight/Az.HDInsight.psd1 (hunk 1 above)
Comment (aim-for-better) [reply src/HDInsight/HDInsight/Az.HDInsight.psd1]: Reverted

---
In reply to: [438664464](https://github.com/Azure/azure-powershell/pull/12124#discussion_r438664464) [](ancestors = 438664464)

**Comment by aim-for-better** [comment]
clock1

**Pull request merged**